pkgs/applications/networking/cluster/k3s/default.nix

```diff
@@ -260,32 +244,74 @@ stdenv.mkDerivation rec {
     conntrack-tools
   ];
 
-  buildInputs = [
-    k3sBin
-  ] ++ k3sRuntimeDeps;
+  buildInputs = k3sRuntimeDeps;
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [
+    makeWrapper
+    rsync
+    yq-go
+    zstd
+  ];
 
-  unpackPhase = "true";
+  # embedded in the final k3s cli
+  propagatedBuildInputs = [
+    k3sCNIPlugins
+    k3sContainerd
+    k3sServer
+    runc
+  ];
+
+  # We override most of buildPhase due to peculiarities in k3s's build.
+  # Specifically, it has a 'go generate' which runs part of the package. See
+  # this comment:
+  # https://github.com/NixOS/nixpkgs/pull/158089#discussion_r799965694
+  # So, why do we use buildGoModule at all? For the `vendorSha256` / `go mod download` stuff primarily.
+  buildPhase = ''
+    # checkPhase expects this to be defined
+    getGoDirs() {
+      :
+    }
+
+    # create a workdir since we need to modify $src, but $src is RO
```

Comment:
Not sure why this is needed, the source in build dir can be written to?

Reply:
Yup, you're right, thanks! I was running into errors that I thought implied otherwise before, but I think I was just doing something else dumb.

Dropped this.